lib/src/rules/always_specify_types.dart

```diff
@@ -149,9 +179,25 @@ class _Visitor extends SimpleAstVisitor<void> {
         param.type == null &&
         !isJustUnderscores(identifier.name)) {
       if (param.keyword != null) {
-        rule.reportLintForToken(param.keyword);
-      } else {
-        rule.reportLint(param);
+        var keyword = param.keyword!;
+        var type = param.declaredElement?.type;
+
+        if (keyword.type == Keyword.VAR &&
+            type != null &&
+            type is! DynamicType) {
+          rule.reportLintForToken(keyword,
+              arguments: [keyword, type], errorCode: keywordCouldBeTypeCode);
+        } else {
+          rule.reportLintForToken(keyword);
```

Comment:
Do we want to use a specific errorCode here?

Reply:
This is related to the below cases, which specific `errorCode` should we use? Since we don't know which type is needed. I guess the default `description` is fine.

```dart
a(var x) {}
d(final x) {}
Foo(var bar);
```

Reply:
The default description is the same as the message for `keywordCouldBeSplitToTypesCode`, but the message for `keywordCouldBeTypeCode` seems more appropriate in this case because there's only one identifier being declared here (though you can't use that code because we don't have a specific type to include in the correction message). Whether you want to define a new type code for this new case is up to you.

But mostly I was asking because it seemed inconsistent to be explicit about some of the codes and to use the default for others. It isn't wrong to do so, and it's OK if you intended to do so, I was just checking that it wasn't on oversight. I should have said as much earlier.

Reply:
> The default description is the same as the message for keywordCouldBeSplitToTypesCode

Not really, the default `correctionMessage` is `Try adding a type annotation.`, but for `keywordCouldBeSplitToTypesCode` it is `splitting the declaration ...`. Both don't need arguments, but the second one is meant for the `var` in the delcaration of different types.

In other words, for a case like `a(var x) {}`, we shouldn't use `keywordCouldBeSplitToTypesCode` as there are not declarations to split.

Reply:
Ok, we can leave it for now.